File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.swift
import Foundation

func solution(_ numbers:[Int], _ direction:String) -> [Int] {
    var num = numbers
    
    if direction == "right" {
        let tmp = num.removeLast()
        return [tmp] + num
    }
    else {
        let tmp = num.removeFirst()
        return num + [tmp]
    }
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.java
import java.util.ArrayDeque;
import java.util.Deque;

class Solution {
    public Deque<Integer> solution(int[] numbers, String direction) {
        int[] answer = {};
        Deque<Integer> deque = new ArrayDeque<>();

        for (int i : numbers)
            deque.add(i);

        if (direction.equals("right")) {
            int tmp = deque.removeLast();
            deque.addFirst(tmp);
        }
        else {
            int tmp = deque.removeFirst();
            deque.addLast(tmp);
        }

        return deque;
    }
}